### DISCUSSION TITLE
Is there some tutorial to migrate my project build by contiki-os to contiki-ng?

### DISCUSSION THREAD

**Discussion by onnonn** [contiki-ng/contiki-ng#1579] [Q&A]
What are the new features in contiki-ng with the comparison of contiki-os? Our project is using si4438 and stm8l151 and stm8s207. Looking forward to your answer！

**Comment by alexrayne** [comment]
contiki-ng not supports 8bit. 32bit and ARM - main focus, and maybe some 16bit.

as for migration - no any special skills and stuff demanded. Make system come a bit simpler, and old code must work fine. Imho, 8bit platforms should builds and work fine  with NG too. But if you wanna ip4 - you in trouble.

there is main changes:
- no 8bit code and platforms
- NG drops `applications`, and some of them are converted to `os/services`. 
- also droped `rime` stack.
- droped ip stack and keep in work only ipv6
- some tools and libs are deployed to sub-repos
- also NG-comunity `not like` inlines, so your 8bit code impacts by lack of optimisation.

i make my own port of old contiki with imported NG [here ](https://github.com/alexrayne/contiki) . it still need some large cleanup from useless stuff lost in hystory, and impacts as repo-size overhead about useless 1GB.

**Comment by atiselsts** [comment]
I'm not aware of any tutorials for code migration, although making one is a good idea!

Thanks Alex for the summary of the differences, quite accurate. Here's the official take from maintainers:

_The aim of Contiki-NG is to eliminate some of Contiki’s limitations in order to enable easier maintenance and quicker evolution.  The main vision is to focus on:
• Standard protocols.  Some of the standards supported by Contiki-NG are IEEE 802.15.4 TSCH, 6LoWPAN, 6TiSCH, RPL, CoAP, MQTT, and LWM2M.
• Support for modern hardware platforms
• Dependability (reliability and security) through modern development practices, continuous integration using simulations and a physical testbed, and security testing techniques_

_Besides these high-level changes, Contiki-NG adds a new configuration and logging system, a new lightweight and reliable RPL implementation (RPL-Lite), and a network administration shell.  It also brings an   extensive cleanup of the codebase, with legacy platforms, protocols and services removed (including IPv4 support)._
